src/shared/codelens/goCodeLensProvider.ts

```diff
@@ -0,0 +1,179 @@
+        // otherwise we would just use symbol.details instead of parsing it ourselves
+        const funcWithBody: string = document.getText(symbol.range)
+
+        const bodyStart: number = funcWithBody.search(/\{/)
+        const sigStart: number = funcWithBody.search(/\(/)
+
+        if (bodyStart === -1 || sigStart === -1) {
+            return false
+        }
+
+        const funcSig: string = stripNewLinesAndComments(funcWithBody.substring(sigStart, bodyStart))
+
+        const funcSigParts: string[] = funcSig.split(/\)/)
+        const argTypes: string[] = parseTypes(funcSigParts[0])
+        const retTypes: string[] = funcSigParts.length > 1 ? parseTypes(funcSigParts[1]) : []
+
+        if (argTypes.length > 2 || retTypes.length > 2) {
+            return false
+        }
+
+        // TODO: actually check the types to make sure they are valid
+
+        return true
+    }
+
+    return false
+}
+
+/**
+ * Finds all types of the parameter list, stripping out names if they exist.
+ * Parenthesis are stripped from the list if they exist.
+ *
+ * @param params List of parameters delimeted by commas
+ *
+ * @returns A list of types mapping 1:1 with the input
+ */
+function parseTypes(params: string): string[] {
+    params = params.replace(/\(|\)/, '')
+    const types: string[] = []
+
+    let lastType: number = 0
+    const paramParts = params.split(',')
+
+    // Names of parameters must either be all present or all absent: https://golang.org/ref/spec#Function_types
+    for (let i = 0; i < paramParts.length; i++) {
+        const parts: string[] = paramParts[i].trim().split(/\s+/)
+        const type: string = parts.length > 1 ? parts[1] : parts[0]
+
+        // Go allows types to be assigned to multiple parameters, e.g. (x, y, z int) === (x int, y int, z int)
+        if (parts.length > 1) {
+            for (let j = lastType; j < i; j++) {
+                types[j] = type
+            }
+            lastType = i + 1
+        }
+
+        types.push(type)
+    }
+
+    return types
+}
+
+/**
+ * Checks if the Go extension exists and is active.
+ */
+async function checkForGoExtension(): Promise<boolean> {
+    const extension = vscode.extensions.getExtension(VSCODE_EXTENSION_ID.go)
+
+    if (extension) {
+        if (extension.isActive) {
+            return true
+        }
+
+        getLogger().debug('Activating extension: %s', VSCODE_EXTENSION_ID.go)
+
+        try {
+            await extension.activate()
+            getLogger().debug('Activated extension: %s', VSCODE_EXTENSION_ID.go)
+
+            return true
+        } catch (err) {
+            if (getLogger().logLevelEnabled('debug')) {
+                getLogger().debug('Failed to activate extension "%s": %O', VSCODE_EXTENSION_ID.go, err as Error)
+            } else {
+                getLogger().info('Failed to activate extension "%s". AWS codelenses will not appear in .go files.')
+            }
```

Comment:
Why different log levels? Is this even necessary? Go extension would probably post a message if it fails, why double up? Potentially annoying for users using the toolkit without the Go extension who accidentally open a Go file.

Reply:
I was wanting to output the stack trace on the debug level but you're right, the Go extension should show an error anyway. I'll log a single message to the debug level instead.